vignettes/errors.Rmd

```diff
@@ -518,6 +536,72 @@ Array extent cannot be determined by time.  This is really a special case of [E2
 dim(a) <- if (time > 10) 100 else 20
 ```
 
+# `E1041`
+
+Special functions on the right-hand-side of an expression must be the only expression there.  This applies to `parameter()`, `interpolate()` and `data()`; if these are used they must be the **only** expression on the right hand side.
+
+For example you cannot write:
+
+```
+x <- interpolate(at, ay) + interpolate(bt, by)
+```
+
+You might write this as:
+
+```
+a <- interpolate(at, ay)
+b <- interpolate(bt, by)
+x <- a + b
+```
+
+# `E1042`
+
+Invalid special function used on the right hand side.  Some odin functions can only be used on the lhs; this include `initial`, `update` and `deriv`.  So you cannot write this:
+
+```
+a <- x + deriv(y)
+```
+
+# `E1043`
+
+Invalid functions used in expression for `dim()`.  Currently `dim()` is quite limited in what it can accept (though slightly more relaxed than `odin1`).  You can use `+`, `-`, `(`, `length`, `nrow` and `ncol` but nothing else.  Let us know if you think you should be able to use more.
+
+# `E1044`
+
+Attempt to call something which is not the **name** of a function.  This might be
+
+```
+a <- 1(x)
+```
+
+where `1` is not a symbol, so this fails.  Other possible errors, which are valid R code but not valid odin code, might include:
+
+```
+a <- f(x)(y)
+```
+
+where `f(x)` might return a function in R, but this is not possible in odin.
+
+# `E1045`
+
+Attempt to use one of R's reserved words: `function`, `while`, `repeat`, `for`, etc.
+
+# `E1046`
+
+All `if` statements must have an `else` clause.  You cannot write:
+
+```
+a <- if (condition) 1
+```
+
+as this is really the same as (in R)
+
+```
+a <- if (condition) 1 else NULL
+```
+
+which would not be valid in R.  Always provide an `else` branch.
```

Comment:
I don't quite get the wording - 
```
a <- if (condition) 1 else NULL
```

is valid in R isn't it?

Reply:
that should be "not valid in **odin**", it is completely valid R yes